src/components/FindMovie/FindMovie.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { TextField } from '@material-ui/core';
+import { Popup } from '../Popup';
+import { getMoviesFromServer } from '../../helpers';
+import './FindMovie.scss';
+
+export const FindMovie = React.memo(({ moviesIds, addNewMovies }) => {
+  const [title, setTitle] = useState('');
+  const [moviesFromServer, setMoviesList] = useState([]);
+  const [error, setError] = useState('');
+
+  const handleUserChoise = (list) => {
+    addNewMovies(list);
+    setMoviesList([]);
+  };
+
+  return (
+    <>
+      <div className="find-movie">
+        <TextField
+          error={!!error}
+          style={{
+            width: 300,
+          }}
+          variant="outlined"
+          label={!error ? 'Movie Title' : error}
+          value={title}
+          onChange={(event) => {
+            setTitle(event.target.value);
+            setError('');
+          }}
+        />
+        <button
+          type="button"
+          className="button"
+          disabled={title.length < 3}
+          onClick={() => {
+            getMoviesFromServer(title, setTitle, setMoviesList, setError);
+          }}
```

Comment:
```suggestion
          onClick={async () => {
            const movies = await getMoviesFromServer(title, setTitle);
            
            if(movies.something) {
              setMoviesList(...)
              return;
            }
            
            setError(...)
          }}
```
Тебе не надо передавать туда все эти функции, жди тут response, и потом решай что тебе надо сделать